From: Michael Froman <mjfroman@mac.com>
Date: Fri, 7 Mar 2025 15:43:47 -0600
Subject: (tmp-cherry-pick) Revert "Move resources to resources/BUILD.gn."
 (e90ab59b7c)

This reverts commit 7dea26d8bb0fbb2f6fe25e74d2baac9293e413a8.

Reason for revert: breaks downstream

Original change's description:
> Move resources to resources/BUILD.gn.
>
> iOS bundle all resources in the same folder and some conflicts can arise from that.
> Having all resources in the same file makes it easier to reason about it.
>
> Change-Id: I37f420dfbd265ec644804e9d4c96515c83d2a992
> Bug: b/397385850
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/377821
> Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
> Commit-Queue: Jeremy Leconte <jleconte@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#43944}

Bug: b/397385850
Change-Id: I80788590498fc24709c95a6a9580fdad65860f8c
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/378280
Commit-Queue: Jeremy Leconte <jleconte@webrtc.org>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Owners-Override: Jeremy Leconte <jleconte@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#43947}
---
 BUILD.gn                          |  14 +-
 common_video/BUILD.gn             |  14 +-
 media/BUILD.gn                    |  22 ++-
 modules/audio_coding/BUILD.gn     |  33 ++++-
 modules/video_coding/BUILD.gn     |  43 +++++-
 pc/BUILD.gn                       |  19 ++-
 resources/BUILD.gn                | 233 ++++--------------------------
 rtc_tools/BUILD.gn                |  19 ++-
 rtc_tools/network_tester/BUILD.gn |  23 ++-
 test/BUILD.gn                     |  22 ++-
 test/pc/e2e/BUILD.gn              |  16 +-
 test/scenario/BUILD.gn            |  29 +++-
 12 files changed, 248 insertions(+), 239 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 96b7d9bead..0d4c96d9b5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -617,6 +617,16 @@ if (use_libfuzzer || use_afl) {
 }
 
 if (rtc_include_tests && !build_with_chromium) {
+  rtc_unittests_resources = [ "resources/reference_video_640x360_30fps.y4m" ]
+
+  if (is_ios) {
+    bundle_data("rtc_unittests_bundle_data") {
+      testonly = true
+      sources = rtc_unittests_resources
+      outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+    }
+  }
+
   rtc_test("rtc_unittests") {
     testonly = true
 
@@ -662,7 +672,7 @@ if (rtc_include_tests && !build_with_chromium) {
       "test/network:network_emulation_unittests",
     ]
 
-    data_deps = [ "resources:rtc_unittests_data" ]
+    data = rtc_unittests_resources
 
     if (rtc_enable_protobuf) {
       deps += [
@@ -672,7 +682,7 @@ if (rtc_include_tests && !build_with_chromium) {
     }
 
     if (is_ios) {
-      deps += [ "resources:rtc_unittests_bundle_data" ]
+      deps += [ ":rtc_unittests_bundle_data" ]
     }
 
     if (is_android) {
diff --git a/common_video/BUILD.gn b/common_video/BUILD.gn
index c7a9e3e01c..1a69762a65 100644
--- a/common_video/BUILD.gn
+++ b/common_video/BUILD.gn
@@ -127,6 +127,16 @@ rtc_source_set("frame_counts") {
 }
 
 if (rtc_include_tests && !build_with_chromium) {
+  common_video_resources = [ "../resources/foreman_cif.yuv" ]
+
+  if (is_ios) {
+    bundle_data("common_video_unittests_bundle_data") {
+      testonly = true
+      sources = common_video_resources
+      outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+    }
+  }
+
   rtc_library("corruption_detection_converters_unittest") {
     testonly = true
     sources = [ "corruption_detection_converters_unittest.cc" ]
@@ -189,13 +199,13 @@ if (rtc_include_tests && !build_with_chromium) {
       "//third_party/libyuv",
     ]
 
-    data_deps = [ "../resources:common_video_data" ]
+    data = common_video_resources
     if (is_android) {
       shard_timeout = 900
     }
 
     if (is_ios) {
-      deps += [ "../resources:common_video_unittests_bundle_data" ]
+      deps += [ ":common_video_unittests_bundle_data" ]
     }
   }
 }
diff --git a/media/BUILD.gn b/media/BUILD.gn
index ad5b0be847..0564eb745b 100644
--- a/media/BUILD.gn
+++ b/media/BUILD.gn
@@ -905,6 +905,24 @@ if (rtc_include_tests) {
   }
 
   if (!build_with_chromium) {
+    rtc_media_unittests_resources = [
+      "../resources/media/captured-320x240-2s-48.frames",
+      "../resources/media/faces.1280x720_P420.yuv",
+      "../resources/media/faces_I400.jpg",
+      "../resources/media/faces_I411.jpg",
+      "../resources/media/faces_I420.jpg",
+      "../resources/media/faces_I422.jpg",
+      "../resources/media/faces_I444.jpg",
+    ]
+
+    if (is_ios) {
+      bundle_data("rtc_media_unittests_bundle_data") {
+        testonly = true
+        sources = rtc_media_unittests_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+
     rtc_test("rtc_media_unittests") {
       testonly = true
 
@@ -1075,14 +1093,14 @@ if (rtc_include_tests) {
         defines += [ "WEBRTC_OPUS_SUPPORT_120MS_PTIME=0" ]
       }
 
-      data_deps = [ "../resources:rtc_media_unittests_data" ]
+      data = rtc_media_unittests_resources
 
       if (is_android) {
         shard_timeout = 900
       }
 
       if (is_ios) {
-        deps += [ "../resources:rtc_media_unittests_bundle_data" ]
+        deps += [ ":rtc_media_unittests_bundle_data" ]
       }
 
       if (rtc_build_dcsctp) {
diff --git a/modules/audio_coding/BUILD.gn b/modules/audio_coding/BUILD.gn
index 681508f3aa..573e7af58a 100644
--- a/modules/audio_coding/BUILD.gn
+++ b/modules/audio_coding/BUILD.gn
@@ -966,6 +966,17 @@ if (rtc_include_tests) {
   }
 
   if (!build_with_chromium) {
+    audio_decoder_unittests_resources =
+        [ "../../resources/audio_coding/testfile32kHz.pcm" ]
+
+    if (is_ios) {
+      bundle_data("audio_decoder_unittests_bundle_data") {
+        testonly = true
+        sources = audio_decoder_unittests_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+
     rtc_test("audio_decoder_unittests") {
       testonly = true
       sources = [ "neteq/audio_decoder_unittest.cc" ]
@@ -988,7 +999,7 @@ if (rtc_include_tests) {
                "//testing/gtest",
              ] + audio_coding_deps
 
-      data_deps = [ "../../resources:audio_decoder_unittests_data" ]
+      data = audio_decoder_unittests_resources
 
       if (is_android) {
         use_default_launcher = false
@@ -996,7 +1007,7 @@ if (rtc_include_tests) {
         shard_timeout = 900
       }
       if (is_ios) {
-        deps += [ "../../resources:audio_decoder_unittests_bundle_data" ]
+        deps += [ ":audio_decoder_unittests_bundle_data" ]
       }
     }
   }
@@ -1058,6 +1069,20 @@ if (rtc_include_tests) {
   }
 
   if (!build_with_chromium) {
+    audio_codec_speed_tests_resources = [
+      "//resources/audio_coding/music_stereo_48kHz.pcm",
+      "//resources/audio_coding/speech_mono_16kHz.pcm",
+      "//resources/audio_coding/speech_mono_32_48kHz.pcm",
+    ]
+
+    if (is_ios) {
+      bundle_data("audio_codec_speed_tests_data") {
+        testonly = true
+        sources = audio_codec_speed_tests_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+
     rtc_test("audio_codec_speed_tests") {
       testonly = true
       defines = []
@@ -1071,7 +1096,7 @@ if (rtc_include_tests) {
         "codecs/tools/audio_codec_speed_test.h",
       ]
 
-      data_deps = [ "../../resources:audio_codec_speed_tests_data" ]
+      data = audio_codec_speed_tests_resources
 
       if (is_android) {
         use_default_launcher = false
@@ -1080,7 +1105,7 @@ if (rtc_include_tests) {
       }
 
       if (is_ios) {
-        deps += [ "../../resources:audio_codec_speed_tests_bundle_data" ]
+        deps += [ ":audio_codec_speed_tests_data" ]
       }
 
       deps += [
diff --git a/modules/video_coding/BUILD.gn b/modules/video_coding/BUILD.gn
index 76323c858f..7e044d26c3 100644
--- a/modules/video_coding/BUILD.gn
+++ b/modules/video_coding/BUILD.gn
@@ -863,6 +863,39 @@ if (rtc_include_tests) {
     ]
   }
 
+  video_coding_modules_tests_resources = []
+  if (is_android) {
+    video_coding_modules_tests_resources += [
+      "../../resources/foreman_128x96.yuv",
+      "../../resources/foreman_160x120.yuv",
+      "../../resources/foreman_176x144.yuv",
+      "../../resources/foreman_240x136.yuv",
+      "../../resources/foreman_320x240.yuv",
+      "../../resources/foreman_480x272.yuv",
+    ]
+  }
+  if (!is_android) {
+    video_coding_modules_tests_resources += [
+      "../../resources/ConferenceMotion_1280_720_50.yuv",
+      "../../resources/FourPeople_1280x720_30.yuv",
+    ]
+  }
+
+  num_video_coding_modules_tests_resources = 0
+  foreach(i, video_coding_modules_tests_resources) {
+    num_video_coding_modules_tests_resources += 1
+  }
+
+  if (num_video_coding_modules_tests_resources > 0) {
+    if (is_ios || is_mac) {
+      bundle_data("video_coding_modules_tests_resources_bundle_data") {
+        testonly = true
+        sources = video_coding_modules_tests_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+  }
+
   rtc_library("videocodec_test_impl") {
     testonly = true
     sources = [
@@ -1050,7 +1083,7 @@ if (rtc_include_tests) {
       "//third_party/libyuv",
     ]
 
-    data_deps = [ "../../resources:video_coding_modules_tests_data" ]
+    data = video_coding_modules_tests_resources
 
     if (is_android) {
       sources += [ "codecs/test/videocodec_test_mediacodec.cc" ]
@@ -1065,10 +1098,10 @@ if (rtc_include_tests) {
       sources += [ "codecs/test/videocodec_test_videotoolbox.cc" ]
 
       deps += [ ":objc_codec_factory_helper" ]
-    }
-    if (is_ios) {
-      deps +=
-          [ "../../resources:video_coding_modules_tests_resources_bundle_data" ]
+
+      if (num_video_coding_modules_tests_resources > 0) {
+        deps += [ ":video_coding_modules_tests_resources_bundle_data" ]
+      }
     }
 
     if (rtc_build_libvpx) {
diff --git a/pc/BUILD.gn b/pc/BUILD.gn
index ac91eaf986..91c36ff9b5 100644
--- a/pc/BUILD.gn
+++ b/pc/BUILD.gn
@@ -2796,8 +2796,24 @@ if (rtc_include_tests && !build_with_chromium) {
     ]
   }
 
+  svc_tests_resources = [
+    "../resources/difficult_photo_1850_1110.yuv",
+    "../resources/photo_1850_1110.yuv",
+    "../resources/presentation_1850_1110.yuv",
+    "../resources/web_screenshot_1850_1110.yuv",
+  ]
+
+  if (is_ios) {
+    bundle_data("svc_tests_bundle_data") {
+      testonly = true
+      sources = svc_tests_resources
+      outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+    }
+  }
+
   rtc_test("svc_tests") {
     sources = [ "test/svc_e2e_tests.cc" ]
+    data = svc_tests_resources
     deps = [
       "../api:create_network_emulation_manager",
       "../api:create_peer_connection_quality_test_frame_generator",
@@ -2827,10 +2843,9 @@ if (rtc_include_tests && !build_with_chromium) {
       "../test/pc/e2e:network_quality_metrics_reporter",
       "../test/pc/e2e/analyzer/video:default_video_quality_analyzer",
     ]
-    data_deps = [ "../resources:svc_tests_data" ]
 
     if (is_ios) {
-      deps += [ "../resources:svc_tests_bundle_data" ]
+      deps += [ ":svc_tests_bundle_data" ]
     }
   }
 }
diff --git a/resources/BUILD.gn b/resources/BUILD.gn
index 71d3500ff2..6d6beab721 100644
--- a/resources/BUILD.gn
+++ b/resources/BUILD.gn
@@ -8,25 +8,6 @@
 
 import("../webrtc.gni")
 
-audio_codec_speed_tests_resources = [
-  "audio_coding/music_stereo_48kHz.pcm",
-  "audio_coding/speech_mono_16kHz.pcm",
-  "audio_coding/speech_mono_32_48kHz.pcm",
-]
-group("audio_codec_speed_tests_data") {
-  data = audio_codec_speed_tests_resources
-}
-
-audio_decoder_unittests_resources = [ "audio_coding/testfile32kHz.pcm" ]
-group("audio_decoder_unittests_data") {
-  data = audio_decoder_unittests_resources
-}
-
-common_video_resources = [ "foreman_cif.yuv" ]
-group("common_video_data") {
-  data = common_video_resources
-}
-
 modules_tests_resources = [
   "audio_coding/testfile16kHz.pcm",
   "audio_coding/testfile32kHz.pcm",
@@ -36,6 +17,13 @@ modules_tests_resources = [
 group("modules_tests_data") {
   data = modules_tests_resources
 }
+if (is_ios) {
+  bundle_data("modules_tests_bundle_data") {
+    testonly = true
+    sources = modules_tests_resources
+    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+  }
+}
 
 modules_unittests_resources = [
   "audio_coding/neteq_opus.rtp",
@@ -143,116 +131,12 @@ modules_unittests_resources = [
 group("modules_unittests_data") {
   data = modules_unittests_resources
 }
-
-network_tester_unittests_resources = [
-  "network_tester/client_config.dat",
-  "network_tester/server_config.dat",
-]
-group("network_tester_unittests_data") {
-  data = network_tester_unittests_resources
-}
-
-peer_connection_e2e_smoke_test_resources = [
-  "pc_quality_smoke_test_alice_source.wav",
-  "pc_quality_smoke_test_bob_source.wav",
-]
-group("peer_connection_e2e_smoke_test_data") {
-  data = peer_connection_e2e_smoke_test_resources
-}
-
-rtc_media_unittests_resources = [
-  "media/captured-320x240-2s-48.frames",
-  "media/faces.1280x720_P420.yuv",
-  "media/faces_I400.jpg",
-  "media/faces_I411.jpg",
-  "media/faces_I420.jpg",
-  "media/faces_I422.jpg",
-  "media/faces_I444.jpg",
-]
-group("rtc_media_unittests_data") {
-  data = rtc_media_unittests_resources
-}
-
-rtc_unittests_resources = [ "reference_video_640x360_30fps.y4m" ]
-group("rtc_unittests_data") {
-  data = rtc_unittests_resources
-}
-
-scenario_resources = [
-  "difficult_photo_1850_1110.yuv",
-  "photo_1850_1110.yuv",
-  "presentation_1850_1110.yuv",
-  "web_screenshot_1850_1110.yuv",
-]
-group("scenario_data") {
-  data = scenario_resources
-}
-
-scenario_unittest_resources = [ "foreman_cif.yuv" ]
-group("scenario_unittest_data") {
-  data = scenario_unittest_resources
-}
-
-svc_tests_resources = [
-  "difficult_photo_1850_1110.yuv",
-  "photo_1850_1110.yuv",
-  "presentation_1850_1110.yuv",
-  "web_screenshot_1850_1110.yuv",
-]
-group("svc_tests_data") {
-  data = svc_tests_resources
-}
-
-test_support_unittests_resources = [
-  "foreman_cif_short.yuv",
-  "video_coding/frame-ethernet-ii.pcap",
-  "video_coding/frame-loopback.pcap",
-  "video_coding/pltype103.rtp",
-  "video_coding/pltype103_header_only.rtp",
-  "video_coding/ssrcs-2.pcap",
-  "video_coding/ssrcs-3.pcap",
-]
-group("test_support_unittests_data") {
-  data = test_support_unittests_resources
-}
-
-tools_unittests_resources = [
-  "foreman_128x96.yuv",
-  "foreman_cif.yuv",
-  "reference_less_video_test_file.y4m",
-  "rtc_event_log/rtc_event_log_500kbps.binarypb",
-]
-group("tools_unittests_data") {
-  data = tools_unittests_resources
-}
-
-video_coding_modules_tests_resources = []
-if (is_android) {
-  video_coding_modules_tests_resources += [
-    "foreman_128x96.yuv",
-    "foreman_160x120.yuv",
-    "foreman_176x144.yuv",
-    "foreman_240x136.yuv",
-    "foreman_320x240.yuv",
-    "foreman_480x272.yuv",
-  ]
-} else {
-  video_coding_modules_tests_resources += [
-    "ConferenceMotion_1280_720_50.yuv",
-    "FourPeople_1280x720_30.yuv",
-  ]
-}
-group("video_coding_modules_tests_data") {
-  data = video_coding_modules_tests_resources
-}
-
-video_engine_tests_resources = [
-  "ConferenceMotion_1280_720_50.yuv",
-  "foreman_cif_short.yuv",
-  "voice_engine/audio_long16.pcm",
-]
-group("video_engine_tests_data") {
-  data = video_engine_tests_resources
+if (is_ios) {
+  bundle_data("modules_unittests_bundle_data") {
+    testonly = true
+    sources = modules_unittests_resources
+    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+  }
 }
 
 webrtc_perf_tests_resources = [
@@ -271,91 +155,26 @@ webrtc_perf_tests_resources = [
 group("webrtc_perf_tests_data") {
   data = webrtc_perf_tests_resources
 }
-
 if (is_ios) {
-  bundle_data("audio_codec_speed_tests_bundle_data") {
-    testonly = true
-    sources = audio_codec_speed_tests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("audio_decoder_unittests_bundle_data") {
-    testonly = true
-    sources = audio_decoder_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("common_video_unittests_bundle_data") {
-    testonly = true
-    sources = common_video_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("modules_tests_bundle_data") {
-    testonly = true
-    sources = modules_tests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("modules_unittests_bundle_data") {
-    testonly = true
-    sources = modules_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("network_tester_unittests_bundle_data") {
-    testonly = true
-    sources = network_tester_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("peer_connection_e2e_smoke_test_resources_bundle_data") {
-    testonly = true
-    sources = peer_connection_e2e_smoke_test_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("rtc_media_unittests_bundle_data") {
-    testonly = true
-    sources = rtc_media_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("rtc_unittests_bundle_data") {
-    testonly = true
-    sources = rtc_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("scenario_resources_bundle_data") {
-    testonly = true
-    sources = scenario_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("scenario_unittest_resources_bundle_data") {
-    testonly = true
-    sources = scenario_unittest_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("svc_tests_bundle_data") {
-    testonly = true
-    sources = svc_tests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("test_support_unittests_bundle_data") {
-    testonly = true
-    sources = test_support_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("tools_unittests_bundle_data") {
-    testonly = true
-    sources = tools_unittests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
-  bundle_data("video_coding_modules_tests_resources_bundle_data") {
+  bundle_data("webrtc_perf_tests_bundle_data") {
     testonly = true
-    sources = video_coding_modules_tests_resources
+    sources = webrtc_perf_tests_resources
     outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
   }
+}
+
+video_engine_tests_resources = [
+  "ConferenceMotion_1280_720_50.yuv",
+  "foreman_cif_short.yuv",
+  "voice_engine/audio_long16.pcm",
+]
+group("video_engine_tests_data") {
+  data = video_engine_tests_resources
+}
+if (is_ios) {
   bundle_data("video_engine_tests_bundle_data") {
     testonly = true
     sources = video_engine_tests_resources
     outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
   }
-  bundle_data("webrtc_perf_tests_bundle_data") {
-    testonly = true
-    sources = webrtc_perf_tests_resources
-    outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
-  }
 }
diff --git a/rtc_tools/BUILD.gn b/rtc_tools/BUILD.gn
index b5183a0cf3..8f8e355b77 100644
--- a/rtc_tools/BUILD.gn
+++ b/rtc_tools/BUILD.gn
@@ -550,6 +550,21 @@ if (rtc_include_tests) {
       }
     }
 
+    tools_unittests_resources = [
+      "../resources/foreman_128x96.yuv",
+      "../resources/foreman_cif.yuv",
+      "../resources/reference_less_video_test_file.y4m",
+      "../resources/rtc_event_log/rtc_event_log_500kbps.binarypb",
+    ]
+
+    if (is_ios) {
+      bundle_data("tools_unittests_bundle_data") {
+        testonly = true
+        sources = tools_unittests_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+
     rtc_test("tools_unittests") {
       testonly = true
 
@@ -597,12 +612,12 @@ if (rtc_include_tests) {
         ]
       }
 
-      data_deps = [ "../resources:tools_unittests_data" ]
+      data = tools_unittests_resources
       if (is_android) {
         shard_timeout = 900
       }
       if (is_ios) {
-        deps += [ "../resources:tools_unittests_bundle_data" ]
+        deps += [ ":tools_unittests_bundle_data" ]
       }
     }
 
diff --git a/rtc_tools/network_tester/BUILD.gn b/rtc_tools/network_tester/BUILD.gn
index 236c99ec31..2ad4abd328 100644
--- a/rtc_tools/network_tester/BUILD.gn
+++ b/rtc_tools/network_tester/BUILD.gn
@@ -62,6 +62,19 @@ if (rtc_enable_protobuf) {
     ]
   }
 
+  network_tester_unittests_resources = [
+    "../../resources/network_tester/client_config.dat",
+    "../../resources/network_tester/server_config.dat",
+  ]
+
+  if (is_ios) {
+    bundle_data("network_tester_unittests_bundle_data") {
+      testonly = true
+      sources = network_tester_unittests_resources
+      outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+    }
+  }
+
   rtc_library("network_tester_unittests") {
     testonly = true
 
@@ -79,13 +92,13 @@ if (rtc_enable_protobuf) {
       "//testing/gtest",
     ]
 
-    defines = [ "WEBRTC_NETWORK_TESTER_TEST_ENABLED" ]
-
-    data_deps = [ "../../resources:network_tester_unittests_data" ]
-
     if (is_ios) {
-      deps += [ "../../resources:network_tester_unittests_bundle_data" ]
+      deps += [ ":network_tester_unittests_bundle_data" ]
     }
+
+    defines = [ "WEBRTC_NETWORK_TESTER_TEST_ENABLED" ]
+
+    data = network_tester_unittests_resources
   }
 
   rtc_executable("network_tester_server") {
diff --git a/test/BUILD.gn b/test/BUILD.gn
index c1f58a213b..598c196bad 100644
--- a/test/BUILD.gn
+++ b/test/BUILD.gn
@@ -653,6 +653,24 @@ if (rtc_include_tests) {
       ]
     }
 
+    test_support_unittests_resources = [
+      "../resources/foreman_cif_short.yuv",
+      "../resources/video_coding/frame-ethernet-ii.pcap",
+      "../resources/video_coding/frame-loopback.pcap",
+      "../resources/video_coding/pltype103.rtp",
+      "../resources/video_coding/pltype103_header_only.rtp",
+      "../resources/video_coding/ssrcs-2.pcap",
+      "../resources/video_coding/ssrcs-3.pcap",
+    ]
+
+    if (is_ios) {
+      bundle_data("test_support_unittests_bundle_data") {
+        testonly = true
+        sources = test_support_unittests_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+
     rtc_library("fixed_fps_video_frame_writer_adapter_test") {
       testonly = true
       sources = [ "testsupport/fixed_fps_video_frame_writer_adapter_test.cc" ]
@@ -782,13 +800,13 @@ if (rtc_include_tests) {
         deps += [ "//third_party/catapult/tracing/tracing:histogram" ]
       }
 
-      data_deps = [ "../resources:test_support_unittests_data" ]
+      data = test_support_unittests_resources
       if (is_android) {
         shard_timeout = 900
       }
 
       if (is_ios) {
-        deps += [ "../resources:test_support_unittests_bundle_data" ]
+        deps += [ ":test_support_unittests_bundle_data" ]
       }
 
       if (!is_android) {
diff --git a/test/pc/e2e/BUILD.gn b/test/pc/e2e/BUILD.gn
index a539be6d1f..92f6b41a4b 100644
--- a/test/pc/e2e/BUILD.gn
+++ b/test/pc/e2e/BUILD.gn
@@ -291,6 +291,18 @@ if (!build_with_chromium) {
       ]
     }
 
+    peer_connection_e2e_smoke_test_resources = [
+      "../../../resources/pc_quality_smoke_test_alice_source.wav",
+      "../../../resources/pc_quality_smoke_test_bob_source.wav",
+    ]
+    if (is_ios) {
+      bundle_data("peer_connection_e2e_smoke_test_resources_bundle_data") {
+        testonly = true
+        sources = peer_connection_e2e_smoke_test_resources
+        outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+      }
+    }
+
     rtc_library("peer_connection_e2e_smoke_test") {
       testonly = true
 
@@ -332,12 +344,12 @@ if (!build_with_chromium) {
         "analyzer/video:default_video_quality_analyzer",
         "analyzer/video:default_video_quality_analyzer_shared",
       ]
-      data_deps = [ "../../../resources:peer_connection_e2e_smoke_test_data" ]
+      data = peer_connection_e2e_smoke_test_resources
       if (is_mac || is_ios) {
         deps += [ "../../../modules/video_coding:objc_codec_factory_helper" ]
       }
       if (is_ios) {
-        deps += [ "../../../resources:peer_connection_e2e_smoke_test_resources_bundle_data" ]
+        deps += [ ":peer_connection_e2e_smoke_test_resources_bundle_data" ]
       }
     }
 
diff --git a/test/scenario/BUILD.gn b/test/scenario/BUILD.gn
index 5346c1171f..f5c5cd5d9b 100644
--- a/test/scenario/BUILD.gn
+++ b/test/scenario/BUILD.gn
@@ -22,6 +22,27 @@ rtc_library("column_printer") {
 }
 
 if (rtc_include_tests && !build_with_chromium) {
+  scenario_resources = [
+    "../../resources/difficult_photo_1850_1110.yuv",
+    "../../resources/photo_1850_1110.yuv",
+    "../../resources/presentation_1850_1110.yuv",
+    "../../resources/web_screenshot_1850_1110.yuv",
+  ]
+  scenario_unittest_resources = [ "../../resources/foreman_cif.yuv" ]
+
+  if (is_ios) {
+    bundle_data("scenario_resources_bundle_data") {
+      testonly = true
+      sources = scenario_resources
+      outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+    }
+    bundle_data("scenario_unittest_resources_bundle_data") {
+      testonly = true
+      sources = scenario_unittest_resources
+      outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
+    }
+  }
+
   rtc_library("scenario") {
     testonly = true
     sources = [
@@ -152,9 +173,9 @@ if (rtc_include_tests && !build_with_chromium) {
     if (rtc_enable_protobuf) {
       deps += [ "../../modules/audio_coding:ana_config_proto" ]
     }
-    data_deps = [ "../../resources:scenario_data" ]
+    data = scenario_resources
     if (is_ios) {
-      deps += [ "../../resources:scenario_resources_bundle_data" ]
+      deps += [ ":scenario_resources_bundle_data" ]
     }
   }
   rtc_library("scenario_unittests") {
@@ -179,9 +200,9 @@ if (rtc_include_tests && !build_with_chromium) {
       "../logging:log_writer",
       "//testing/gmock",
     ]
-    data_deps = [ "../../resources:scenario_unittest_data" ]
+    data = scenario_unittest_resources
     if (is_ios) {
-      deps += [ "../../resources:scenario_unittest_resources_bundle_data" ]
+      deps += [ ":scenario_unittest_resources_bundle_data" ]
     }
   }
 }
